### DISCUSSION TITLE
Screenshots

### DISCUSSION THREAD

**Discussion by suntong** [MontFerret/ferret#659] [Q&A]
> Output of the function is base64-encoded binary file

I tried it and got "`"/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAA...`"

and when I echo it to `base64 -d`, i.e.,

`echo "/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUAAQEAAAIYAAAAAAIQAABtbnRyUkdCIFhZWi...AiIgCIiAIiIAiIgCIiAIiIAiIgCIiAIiIAiIgCIiAIiIAiIgCIiAIiIAiIgCIiAIiIAiIgCIiAIiIAiIgCIiALyYX/VkT/ulj/wBbgiLMetfxX9QeMvuAIiis9X8/9CSv1/y/1CIijJAiIgCIiAIiIAiIgCIiAIiIAiIgCIiA/9k=" | base64 -d`

I got

`base64: invalid input`

**Comment by ziflex** [comment]
Hey, could you share your query?
It's hard to say what could go wrong without seen an actual query.

**Comment by suntong** [comment]
I just grab the output from https://www.montferret.dev/cookbook/screenshot/
then pass it to `echo` and `base64 -d` as shown above.

I tried again, and the output is still
`"/9j/4AAQSkZJRgABAQAAAQABAAD/4gIoSUNDX1BST0ZJTEUA...`

**Comment by ziflex** [comment]
@suntong hey, the problem is that the output is a JSON string which means it's wrapped with double quotes.
If you remove the quotes and pass it to ``base64`` it will work.

Try this:

```
ferret exec ./examples/screenshot.fql | sed 's/^.\(.*\).$/\1/' | base64 -d > screenshot.jpeg
```
